test/libsolidity/syntaxTests/abiEncoder/abi_encodeCall_tuple_from_assignment_expression.sol

```diff
@@ -0,0 +1,28 @@
+contract C {
+    function g0() internal {}
+    function g1() internal returns (uint) { return (1); }
+    function g2() internal returns (uint, uint) { return (2, 3); }
+
+    function f0() public {}
+    function f1(uint) public {}
+    function f2(uint, uint) public {}
+
+    function h() public view {
+        uint a;
+        uint b;
+
+        abi.encodeCall(this.f0, () = g0());
+        abi.encodeCall(this.f0, () = ());
+        abi.encodeCall(this.f1, (a) = g1()); // Ok
+        abi.encodeCall(this.f1, (a) = (1)); // Ok
```

Comment:
Oh, it's actually weird that we allow that but I guess that's a consequence of the fact that a single-element tuple is not really a tuple...

Reply:
Yeah, in the end it becomes the type of the only element, right?

Reply:
Yes. I mean, I think there is some AST annotation that lets you detect that it's wrapped in the parentheses. I remember @Marenz having to deal with that for weird cases where stuff could be wrapped (e.g. calls like `(L.f)()`).

But I'd say that dealing with that is out of scope of this PR. If we decided to do this, we'd want to change this for all the encoding functions. It would also be breaking because it would disallow some cases that now work fine.